tests/legacy/test_read_exceptions.py

```diff
@@ -203,3 +206,80 @@ def test_timeout_stream(self, generate_task, real_x_series_device):
         assert not any(element == sentinel_value for element in data[:number_of_samples_expected])
         # The data that wasn't read should have been unmodified.
         assert all(element == sentinel_value for element in data[number_of_samples_expected:])
+
+    @pytest.mark.grpc_xfail(reason="Requires NI gRPC Device Server version 2.2 or later")
+    def test_multichannel_timeout_stream(self, generate_task, real_x_series_device):
+        """Test for validating read timeout."""
+        # USB streaming is very tricky.
+        if not (
+            real_x_series_device.bus_type == BusType.PCIE
+            or real_x_series_device.bus_type == BusType.PXIE
+        ):
+            pytest.skip("Requires a plugin device.")
+
+        samples_to_read = 75
+        clocks_to_give = 100
+        sample_rate = 1000
+        # Choose a sentinel value that can't be returned by the read.
+        sentinel_value = -100.0
+        number_of_channels = 2
+
+        read_task = generate_task()
+        sample_clk_task = generate_task()
+        # Use a counter output pulse train task as the sample clock source
+        # for both the AI and AO tasks.
+        sample_clk_task.co_channels.add_co_pulse_chan_freq(
+            f"{real_x_series_device.name}/ctr0", freq=sample_rate, idle_state=Level.LOW
+        )
+        sample_clk_task.timing.cfg_implicit_timing(
+            samps_per_chan=clocks_to_give, sample_mode=AcquisitionType.FINITE
+        )
+        sample_clk_task.control(TaskMode.TASK_COMMIT)
+
+        samp_clk_terminal = f"/{real_x_series_device.name}/Ctr0InternalOutput"
+
+        channel_names = f"{real_x_series_device.ai_physical_chans[0].name},{real_x_series_device.ai_physical_chans[1].name}"
+        read_task.ai_channels.add_ai_voltage_chan(channel_names, max_val=10, min_val=-10)
+        read_task.timing.cfg_samp_clk_timing(
+            sample_rate, source=samp_clk_terminal, sample_mode=AcquisitionType.CONTINUOUS
+        )
+
+        reader = nidaqmx.stream_readers.AnalogMultiChannelReader(read_task.in_stream)
+
+        # Start the read task before the clock.
+        read_task.start()
+
+        # Generate all the clocks.
+        sample_clk_task.start()
+        sample_clk_task.wait_until_done()
+        sample_clk_task.stop()
+
+        # Set read timeout.
+        read_task.in_stream.timeout = 2
+
+        # Do a partial read which should succeed.
+        data = numpy.full(
+            (number_of_channels, samples_to_read), sentinel_value, dtype=numpy.float64
+        )
+        num_values_read = reader.read_many_sample(
+            data, number_of_samples_per_channel=samples_to_read, timeout=2.0
+        )
+        assert num_values_read == samples_to_read
+        # All the data should have been overwritten.
+        assert (data != sentinel_value).any()
```

Comment:
This will be addressed in  [Bug 2420742](https://dev.azure.com/ni/DevCentral/_workitems/edit/2420742): gRPC interpreter doesn't retain the initial array values when converting flattened array into numpy array.